encoding/gzip/gzip.go

```diff
@@ -0,0 +1,56 @@
+package gzip
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"sync"
+
+	"google.golang.org/grpc/encoding"
+)
+
+// RegisterNewGIZPCompressor registers gzip compressor.
+func RegisterNewGIZPCompressor() {
```

Comment:
Done. gzip.New() returns an encoding.Compressor. gzip.compressor has 2 pools like the old compressor and decompressor.